src/framework/theme/components/search/search.component.ts

```diff
@@ -149,6 +151,12 @@ export class NbSearchFieldComponent implements OnChanges, AfterViewInit {
     }
   }
 
+  submitSearchOnInput(term: string) {
+    if (term) {
```

Comment:
It's better to behave like native input event, to not confuse people already familiar with it. So if native event emits when the value becomes empty, we should do this as well.